src/Context/HistoryContext.jsx

```diff
@@ -0,0 +1,82 @@
+import { createContext, useContext, useReducer, useState } from "react";
+import axios from "axios";
+import { useAlert } from "./AlertContext";
+import { v4 as uuid } from "uuid";
+import { actionAddVideoToHistory, actionToDeleteVideoFromHistory, actionToClearAllHistory } from "../Helpers/Services/actions";
+
+
+//CONTEXT CREATED
+const HistoryContext = createContext();
+
+const HistoryProvider = ({ children }) => {
+
+    const encodedToken = localStorage.getItem("authData");
+    const [historyVideos , sethistoryVideos] = useState([]);
+    const {alertContent , setAlertContent} = useAlert();
+
+    //API - actionAddVideoToPlaylist
+    const addVideoToHistory = (videoObj) =>{
+        actionAddVideoToHistory({video:videoObj})
+      .then(res=> {
+          if(res.status === 201 || res.status === 200){
+            sethistoryVideos(res?.data?.history);
```

Comment:
What is the use of  "?" after res and .data

Reply:
i think it checks if the value is defined or not before proceeding its called optional chaining.